### DISCUSSION TITLE
23/04/18 RobotLibrary Development Meeting

### DISCUSSION THREAD

**Discussion by ssutjipto** [Woolfrey/software_robot_library#88] [Meetings]
### Chair:
- Sheila @ssutjipto 

### Attendees:
- Jon @Woolfrey 

### Agenda:
| Item | Tag | Responsible | Comment |
|------|-----|---------------|------------|
| Create Issues/Milestones for the tasks below and tag them | N/A | @Woolfrey | Tags created |
| Create feature(?) branch for `KinematicTree` class. | N/A | @Woolfrey | Branch created |
| Finish testing Trajectory class with ROS2 | N/A | @Woolfrey | Busy with ergoCub, reworking theory on orientation trajectory |
| Replace `Eigen::Isometry3f` with `Pose` in `RigidBody` class. | #76 | @Woolfrey | Done, but need to test product operator |
| Start on parsing URDF files | #77 | @ssutjipto | Done, currently being converted into class by @Woolfrey |
| Insert `Link` properties from `URDF` in to a `Branch` class. | #78 | @ssutjipto | Done, currently being converted into class by @Woolfrey |
| Develop skeleton for `KinematicTree` class | #83  | @Woolfrey | Ongoing |

### Additional Discussion Points:
| Discussion Point | Raised By | Tag | Outcome |
|-------------------|------------|-----|------------|
| Orientation trajectories | @Woolfrey | | Now using angle x axis, contained in `CartesianTrajectory` class |
| Exception handling | @Woolfrey | | Throw exception on constructors, return boolean where possible in getters |
| Naming convention | @ssutjipto | | Underscores for functions and camel case for variables, style guide should be written |
| Combining links with fixed joints | @Woolfrey | | Last step before `KinematicTree` constructor is complete |

### Next Meeting:
| Item | Tag | Responsible |
|------|-----|---------------|
| Close off outstanding issues | #85 #86 | @Woolfrey |
| Finish `KinematicTree` constructor | | @Woolfrey |
| Check `Pose` class | | @ssutjipto |

### Conclusions
Let's try to make this year a good one!

**Comment by Woolfrey** [comment]
![image](https://user-images.githubusercontent.com/62581255/232708554-63a4f85b-302b-4c39-a5a2-cf4fbeff5d4f.png)
